File: Pet.Tenant/Tenant.cs
using System;

namespace Pet.Tenant
{
    public class Tenant
    {
        public Guid TenantId { get; }
        public string ConnectionString { get; }

        public Tenant(Guid tenantId, string connectionString)
        {
            TenantId = tenantId;
            ConnectionString = connectionString;
        }
    }
}

File: Pet.Web/ClientApp/src/components/Tenant.js
import React from 'react';

export default class Tenant extends React.PureComponent {
    componentDidMount() {
        // This method is called when the component is first added to the document
        this.storeTenantId();
    }

    componentDidUpdate(prevProps) {
        // This method is called when the route parameters change
        if (this.props.match.params.tenantId !== prevProps.match.params.tenantId) {
            this.storeTenantId();
        }
    }

    storeTenantId() {
        const tenantId = this.props.match.params.tenantId;
        localStorage.setItem('TenantId', tenantId);
    }

    render() {
        const tenantId = this.props.match.params.tenantId;
        return (
            <div>
                <p>Tenant {tenantId}</p>
            </div>
        )
    }
}